core/src/main/java/com/linecorp/armeria/server/HttpServerUpgradeHandler.java

```diff
@@ -75,8 +79,14 @@ final class HttpServerUpgradeHandler extends ChannelInboundHandlerAdapter {
 
     private static final FullHttpResponse UPGRADE_RESPONSE = newUpgradeResponse();
 
+    private static final ByteBuf INVALID_SETTINGS_HEADER_MESSAGE = Unpooled.unreleasableBuffer(
+            Unpooled.directBuffer()
+                    .writeBytes("Invalid HTTP2-Settings header\n".getBytes(StandardCharsets.UTF_8)));
+
     private static final Splitter COMMA_SPLITTER = Splitter.on(',').omitEmptyStrings().trimResults();
 
+    private final FullHttpResponse invalidSettingsHeaderResponse = newInvalidSettingsHeaderResponse();
```

Comment:
How about removing this field and creating this response at line 293?
This `invalidSettingsHeaderResponse` is not used if `prepareUpgradeResponse` is `true` and it's used only once when `prepareUpgradeResponse` returns `true`.
We don't need this as a member.

Reply:
Had a chat with @trustin and let's do:
- make this field static and inline method
```suggestion
    private static final FullHttpResponse invalidSettingsHeaderResponse =
            new DefaultFullHttpResponse(HTTP_1_1, BAD_REQUEST, INVALID_SETTINGS_HEADER_MESSAGE.duplicate());
```
- Use `duplicate` when it's used. `invalidSettingsHeaderResponse.retain()`